packages/firestore/src/platform/connection.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isNode, isReactNative } from '@firebase/util';
+import {
```

Comment:
This is pretty hefty amount of code to make all these distinct imports. If this is what it takes I guess this is what we're stuck with, but it doesn't seem like this file needs to be tree-shakeable so whole module imports seem like they'd work and would cost way less.

Reply:
It is pretty ugly indeed. I reverted back to namespace imports.